### DISCUSSION TITLE
Is possible to click the arrow in the "DropDownButton" and not fire the "Click" event？

### DISCUSSION THREAD

**Discussion by LiveIsLive** [MahApps/MahApps.Metro#4314] [Q&A]
I find that the "Click" event of the "DropDownButton" will be fired when I click any part of the button in the "DropDownButton", even if I just want to click the arrow to open the dropdown menu.

Is it possible to implementation that when I click the arrow , the "Click" event will not be fired and open the dropdown menu. And when I click button area outside the arrow, the "Click" event will be fired and not open the dropdown menu？

**Comment by timunie** [comment]
Iirc there is a split button which does what you are looking for. You may want to change the style if you want it to look 1:1 like a drop down button, but imo its better to leave it as it is.

**Reply by LiveIsLive** [reply]
I have a try to change the DropDownButton to a SplitButton, but the Text of the SplitButton must be the selection of its ItemsSource.
For Example, the Text property of the SplitButton is not work in the code below.

```
<StackPanel xmlns:mah="http://metro.mahapps.com/winfx/xaml/controls" xmlns:s="clr-namespace:System;assembly=mscorlib">
	<mah:SplitButton Text="Open">
		<mah:SplitButton.ItemsSource>
			<x:Array Type="{x:Type s:String}">
				<s:String>Audio</s:String>
				<s:String>Video</s:String>
				<s:String>Document</s:String>
			</x:Array>
		</mah:SplitButton.ItemsSource>
	</mah:SplitButton>
	<mah:DropDownButton Content="Open">
		<mah:DropDownButton.ItemsSource>
			<x:Array Type="{x:Type s:String}">
				<s:String>Audio</s:String>
				<s:String>Video</s:String>
				<s:String>Document</s:String>
			</x:Array>
		</mah:DropDownButton.ItemsSource>
	</mah:DropDownButton>
</StackPanel>
```

**Reply by timunie** [reply]
You can bind Content to selected item for example. Or handle it via MVVM. That way you even have more control about what is going on imo.

**Reply by LiveIsLive** [reply]
> You can bind Content to selected item for example. Or handle it via MVVM. That way you even have more control about what is going on imo.

I don't want the text of the button display in the dropdown menu. they are different properties in the ViewModel. The DropDownButton can be defined the text of button and have nothing to do with the ItemsSource and selected item.

**Reply by timunie** [reply]
Yes that should be possible I think. I have done it the other day, but can't remember the solution I had. Maybe it also was two buttons next to each other.
